include/seqan3/range/detail/inherited_iterator_base.hpp

```diff
@@ -171,137 +171,154 @@ class inherited_iterator_base : public std::conditional_t<std::is_pointer_v<base
     }
     //!\}
 
-    /*!
-ame Arithmetic operators
+    /*!\name Arithmetic operators
      * \brief Unless specialised in derived_type, all operators perform base_t's operator and cast to derived_t.
      * \{
     */
     //!\brief Pre-increment, return updated iterator.
+    //!\cond
+    template <typename base_t_ = base_t>
+    //!\endcond
     constexpr derived_t & operator++() noexcept(noexcept(++std::declval<base_t &>()))
     //!\cond
-        requires std::input_iterator<base_t>
+        requires requires (base_t_ i) { ++i; }
```

Comment:
Well, it certainly is unlikely that there is an iterator that does not have implemented the ++iterator, but in case it does your code will fail with a hard error instead of rejecting the function because of an unsatisfied constrained. That is because base_t is not a template argument of the function and thus it fails to compile if it does not support the ++ interface. The other solution would be to define for every operator a specific concept. But I really prefer the current solution a lot more.